### PR TITLE
support xpu device

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

feature

## Description

support xpu device quantization

## Expected Behavior & Potential Risk

cannot work with resnet50 now, JIRA is raised to XPU team.

### CODE REVIEW

**Comment by thuang6** [comment]
@xin3he, would you please post the jira link and better to check their fix plan with IPEX team.

File: neural_compressor/config.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ def device(self):
 
     @device.setter
     def device(self, device):
-        if _check_value("device", device, str, ["cpu", "gpu", "npu"]):
+        if _check_value("device", device, str, ["cpu", "gpu", "xpu"]):
```
Comment (chensuyue) [inline neural_compressor/config.py:367]: npu support in onnxrt.

File: examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/main.py [hunk 2]
```diff
@@ -106,6 +106,8 @@
                     help='run benchmark')
 parser.add_argument('--ipex', dest='ipex', action='store_true',
                     help='tuning or benchmark with Intel PyTorch Extension')
+parser.add_argument('--xpu', action='store_true',
+                    help='whether use xpu')
```
Comment (changwangss) [inline examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/main.py:110]: Please add README how to use and how to install xpu.
Could you unity script(cpu, gpu, xpu) by args parser, the path show "cpu", is it limitation?

Comment (xin3he) [reply examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/main.py:110]: This xpu args is for internal test. @violetch24 will take own of examples enabling.
The env building of XPU is not ready to share due to lots of local resources are used.

File: neural_compressor/config.py [hunk 3]
```diff
@@ -756,7 +756,7 @@ class _BaseQuantizationConfig:
                  'optypes_to_exclude_output_quant': don't quantize output of specified optypes
                  'dedicated_qdq_pair': whether dedicate QDQ pair, only valid for onnxrt_trt_ep
         quant_format: Support 'default', 'QDQ' and 'QOperator', only required in ONNXRuntime.
-        device: Support 'cpu', 'gpu' and 'npu'.
+        device: Support 'cpu', 'gpu' and 'xpu'.
```
Comment (chensuyue) [inline neural_compressor/config.py:759]: fix the doc

File: neural_compressor/config.py [hunk 4]
```diff
@@ -1196,7 +1196,7 @@ class PostTrainingQuantConfig(_BaseQuantizationConfig):
     """Config Class for Post Training Quantization.
 
     Args:
-        device: Support 'cpu', 'gpu' and 'npu'.
+        device: Support 'cpu', 'gpu' and 'xpu'.
```
Comment (chensuyue) [inline neural_compressor/config.py:1199]: fix the doc

File: neural_compressor/config.py [hunk 5]
```diff
@@ -1381,7 +1381,7 @@ class QuantizationAwareTrainingConfig(_BaseQuantizationConfig):
     """Config Class for Quantization Aware Training.
 
     Args:
-        device: Support 'cpu', 'gpu' and 'npu'.
+        device: Support 'cpu', 'gpu' and 'xpu'.
```
Comment (chensuyue) [inline neural_compressor/config.py:1384]: fix the doc

File: neural_compressor/config.py [hunk 6]
```diff
@@ -1869,7 +1869,7 @@ class MixedPrecisionConfig(object):
 
     Args:
         device (str, optional): Device for execution.
-                                Support 'cpu', 'gpu' and 'npu', default is 'cpu'.
+                                Support 'cpu', 'gpu' and 'xpu', default is 'cpu'.
```
Comment (chensuyue) [inline neural_compressor/config.py:1872]: fix the doc

**Comment by chensuyue** [comment]
@violetch24 @xin3he any progress about distilbert support?

**Review by chensuyue — APPROVED** [review]

**Pull request merged**